### PR TITLE
Iterate design on several screens: update icons, replace PreferenceTexts

### LINKED ISSUE #1718: Update icons
### Steps to reproduce

I'll add a load of screenshots to communicate the changes. Any icon with a blue dot next to it is already in the Compound icon repo, otherwise it's in a ZIP file which I'll add to this issue:

Screenshot 2023-11-01 at 11 48 29

Screenshot 2023-11-01 at 11 48 34

Screenshot 2023-11-01 at 11 48 38

Screenshot 2023-11-01 at 11 48 51

Screenshot 2023-11-01 at 11 49 09

Screenshot 2023-11-01 at 11 49 20

Screenshot 2023-11-01 at 11 49 23

Screenshot 2023-11-01 at 11 49 46

[Location sharing pin from Figma](https://www.figma.com/file/gkNXqPoiJhEv2wt0EJpew4/Compound-Icons?type=design&node-id=2830%3A3523&mode=design&t=QmgXVm7dBK7ZQKTu-1)

ZIP file with other icons: 
[New icons November 2023.zip](https://github.com/vector-im/element-x-android/files/13226542/New.icons.November.2023.zip)

### Your phone model

_No response_

### Operating system version

_No response_

### Application version and app store

_No response_

### Homeserver

_No response_

### Will you send logs?

Yes

### Are you willing to provide a PR?

Yes

### PR DESCRIPTION
**Pull request by jmartinesp**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Technical
- [x] Other : update icons and usage of Compound components in preferences.

## Content

- Batch import new icons from the design team.
- Rename `_september` icons since they're just extra icons that need to be integrated in Compound in the future, and it should be ok if we don't distinguish between `ic_september_*`, `ic_november_*` etc., so all icons are now simply `ic_*` in the designsystem module.
- Create a new `CompoundIconListPreviewProvider` to add chunked lists of icons for previews. Add an exception for it to Konsist.
- Move some icons to use Compound icons.
- Remove most `PreferenceText` usages, use `ListItem` instead.

## Motivation and context

Closes #1718 .

## Screenshots / GIFs

There are *lots* of screenshot updates in the PR.

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s): 11

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [x] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR

### CODE REVIEW

**Review by ganfra — APPROVED** [review]
LGTM, you just have screenshot conflicts...

**Comment by jmartinesp** [comment]
@callumu could you check if the icons and colors match the designs before merging this?

**Comment by callumu** [comment]
> @callumu could you check if the icons and colors match the designs before merging this?

Yes I can look now. 

Just to confirm - have you updated all icons?

**Comment by callumu** [comment]
> > Just to confirm - have you updated all icons?
> 
> Yes, all of them should be up to date.

Okay thanks. I've just tested and I have one comment:

- For the screen lock icon in Settings, can we use [this one](https://www.figma.com/file/gkNXqPoiJhEv2wt0EJpew4/Compound-Icons?type=design&node-id=2867%3A3668&mode=design&t=jw19AsowF4v449hG-1)?

**Comment by callumu** [comment]
> > > Just to confirm - have you updated all icons?
> > 
> > 
> > Yes, all of them should be up to date.
> 
> Okay thanks. I've just tested and I have one comment:
> 
> * For the screen lock icon in Settings, can we use [this one](https://www.figma.com/file/gkNXqPoiJhEv2wt0EJpew4/Compound-Icons?type=design&node-id=2867%3A3668&mode=design&t=jw19AsowF4v449hG-1)?

And on the 'Report a problem' page, can we change the header to 'Report a problem' and the copy inside the text field to 'Describe the problem...'?

**Comment by callumu** [comment]
> > > > Just to confirm - have you updated all icons?
> > > 
> > > 
> > > Yes, all of them should be up to date.
> > 
> > 
> > Okay thanks. I've just tested and I have one comment:
> > 
> > * For the screen lock icon in Settings, can we use [this one](https://www.figma.com/file/gkNXqPoiJhEv2wt0EJpew4/Compound-Icons?type=design&node-id=2867%3A3668&mode=design&t=jw19AsowF4v449hG-1)?
> 
> And on the 'Report a problem' page, can we change the header to 'Report a problem' and the copy inside the text field to 'Describe the problem...'?

Can we also remove the icon inside the Notifications Settings page?

The reason is to have a consistent rule to show icons on the initial settings page only, not as you navigate to other pages

File: tests/uitests/src/test/snapshots/images/ui_S_t[f.location.impl.send_SendLocationView_null_SendLocationView-Night-0_1_null_4,NEXUS_5,1.0,en].png
Comment (bmarty) [inline tests/uitests/src/test/snapshots/images/ui_S_t[f.location.impl.send_SendLocationView_null_SendLocationView-Night-0_1_null_4,NEXUS_5,1.0,en].png:1]: Just scrolling among the list of changes and seeing this. Is it expected?

File: tests/uitests/src/test/snapshots/images/ui_S_t[f.preferences.impl.root_PreferencesRootViewLight_null_PreferencesRootViewLight--0_2_null_0,NEXUS_5,1.0,en].png
Comment (bmarty) [inline tests/uitests/src/test/snapshots/images/ui_S_t[f.preferences.impl.root_PreferencesRootViewLight_null_PreferencesRootViewLight--0_2_null_0,NEXUS_5,1.0,en].png:1]: We may want the red indicator and the `external` icon to be aligned vertically?

**Comment by jmartinesp** [comment]
> And on the 'Report a problem' page, can we change the header to 'Report a problem' and the copy inside the text field to 'Describe the problem...'?

About this, I think @bmarty started a discussion about changing all these strings, but I can't remember how it ended. Should we create new ones an replace the existing ones, or should we just modify their contents and ask for new translations?

File: tests/uitests/src/test/snapshots/images/ui_S_t[f.location.impl.send_SendLocationView_null_SendLocationView-Night-0_1_null_4,NEXUS_5,1.0,en].png
Comment (jmartinesp) [reply tests/uitests/src/test/snapshots/images/ui_S_t[f.location.impl.send_SendLocationView_null_SendLocationView-Night-0_1_null_4,NEXUS_5,1.0,en].png:1]: This should be fixed now.

**Comment by bmarty** [comment]
> Should we create new ones an replace the existing ones, or should we just modify their contents and ask for new translations?

I guess you can use `common_report_a_problem` now that it exists. For others instance of `bug` in the wordings, less follow the rules:

image

Change the value in Localazy for:
- `screen_bug_report_editor_description`
- `screen_bug_report_editor_placeholder`
- `rageshake_detection_dialog_content`
- `preference_rageshake`

Create new strings on Localazy for:
- `action_report_bug`: create `action_report_a_problem`

Maybe do that in a dedicated PR, this (big) one can be merged now.

**Comment by jmartinesp** [comment]
Maestro passed [here](https://github.com/vector-im/element-x-android/actions/runs/6875187159/job/18698285975), so I'm merging this.

**Pull request merged**